src/personalnotification.cpp

```diff
@@ -36,28 +36,39 @@ using namespace CommHistory;
 // events
 struct EventTypes {
     int type;
-    const char* event;
+    QLatin1String event;
+    QLatin1String category;
+    bool both;
 };
 
 static const EventTypes _eventTypes[] =
 {
-    {CommHistory::Event::IMEvent,       "x-nemo.messaging.im"},
-    {CommHistory::Event::SMSEvent,      "x-nemo.messaging.sms"},
-    {CommHistory::Event::MMSEvent,      "x-nemo.messaging.mms"},
-    {CommHistory::Event::CallEvent,     "x-nemo.call.missed"},
-    {CommHistory::Event::VoicemailEvent,"x-nemo.messaging.voicemail"},
-    {VOICEMAIL_SMS_EVENT_TYPE,          "x-nemo.messaging.voicemail-SMS"}
+    {CommHistory::Event::IMEvent,       QLatin1String("chat"),QLatin1String("x-nemo.messaging.im"), true },
+    {CommHistory::Event::SMSEvent,      QLatin1String("sms"), QLatin1String("x-nemo.messaging.sms"),  true },
+    {CommHistory::Event::MMSEvent,      QLatin1String("sms"), QLatin1String("x-nemo.messaging.mms"),  true },
+    {CommHistory::Event::CallEvent,     QLatin1String("call"),QLatin1String("x-nemo.call.missed"), false},
+    {CommHistory::Event::VoicemailEvent,QLatin1String("sms"), QLatin1String("x-nemo.messaging.voicemail"),  true },
+    {VOICEMAIL_SMS_EVENT_TYPE,          QLatin1String("sms"), QLatin1String("x-nemo.messaging.voicecmail-SMS"),  true }
 };
 
 static const int _eventTypesCount = sizeof(_eventTypes) / sizeof(EventTypes);
 
-static QString groupType(int eventType)
+static void applyTypes(Notification *notification, int eventType, bool restore)
 {
     for (int i = 0; i < _eventTypesCount; i++) {
-        if (_eventTypes[i].type == eventType)
-            return QLatin1String(_eventTypes[i].event);
+        if (_eventTypes[i].type == eventType) {
+            QString feedbackType;
+            if (!_eventTypes[i].both || restore) {
+                feedbackType = QStringLiteral("%1_exists").arg(_eventTypes[i].event);
+            } else {
+                feedbackType = QStringLiteral("%1,%1_exists").arg(_eventTypes[i].event);
+            }
+            notification->setHintValue("x-nemo-feedback", feedbackType);
+            notification->setCategory(_eventTypes[i].category);
```

Comment:
What I was expecting was getting rid of the category files and just applying the notification hints in c++. Why still use these?

Reply:
Better?

Reply:
Starts looking nice.

Reply:
I guess this is done then.